docs/architecture.rst

```diff
@@ -1,66 +1,167 @@
 
-The below diagram provides a high-level overview of the dependencies between ``tqec``
+The diagram below provides a high-level overview of the dependencies between ``tqec``
 submodules. It is not 100% accurate, as for example the ``tqec.circuit`` submodule
 defines a few core data-structures that are used in ``tqec.plaquette``, but this is
 the order in which the sub-modules will be covered here.
 
 .. mermaid::
+    :config: {"theme": "base", "darkMode": "true"}
+
+    %%{
+    init: {
+        'theme': 'base',
+        'themeVariables': {
+        'primaryColor': '#AFEEEE',
+        'lineColor': '#6495ED'
+        }
+    }
+    }%%
 
     graph
         A[tqec.interop] --> B[tqec.computation]
         B --> E[tqec.compile]
         C[tqec.templates] --> E
         D[tqec.plaquette] --> E
         E --> F[tqec.circuit]
-        F --> G[tqec.noise_models]
+        F --> G[tqec.utils.noise_models]
         G --> H[tqec.simulation]
 
-``tqec.interop``
-----------------
+A user starts with a computation and an observable. The input computation can be provided as a :class:`.BlockGraph`
+structure while an observable is represented as a correlation surface. The end goal of ``tqec`` is to generate a ``stim.Circuit``
+such that the input is transformed into a topologically quantum error corrected quantum computation protected by a surface code.
 
-TODO
+.. _interop_ref:
 
-``tqec.computation``
+:mod:`.interop`
 --------------------
 
-TODO
+Allows a user to provide a ``.dae`` file or a ``pyzx`` graph as an input to ``tqec``.
 
-``tqec.compile``
---------------------
+* A :class:`.BlockGraph` can be constructed through a Collada ``.dae`` file through :func:`.read_block_graph_from_dae_file`
+  and vice versa through :func:`.write_block_graph_to_dae_file`.
+* A ZX graph can be mapped to a :class:`.BlockGraph` through :class:`.SynthesisStrategy`.
 
-TODO
+Standard color representations as described in :ref:`terminology` are predefined in :class:`.TQECColor`.
 
-``tqec.templates``
-------------------
+Some conventional implementations of computations in ``.dae`` format are available in the :ref:`gallery-reference-label`.
 
-TODO
+:mod:`.computation`
+--------------------
+
+Defines data structures for the high-level :class:`.BlockGraph` representations of a fault-tolerant computation protected by the surface code.
+
+The ``3D`` structures discussed in detail in :ref:`terminology` are defined in this module.
 
-``tqec.plaquette``
-------------------
+* A :class:`.CorrelationSurface` represents a set of parity measurements between the input and output logical operators.
+* A :class:`.Cube` is a fundamental building block constituting of a block of quantum operations that occupy a specific spacetime volume.
+  Quantum information encoded in the logical qubits can be preserved or manipulated by these blocks.
+* A :class:`.Pipe` is a block that connects :class:`.Cube` objects in a :class:`.BlockGraph` but does not occupy spacetime volume on its own.
```

Comment:
About that, now temporal hadamard pipes have a volume when compiled using the fixed bulk convention.